dbt_common/clients/system.py

```diff
@@ -35,6 +37,7 @@
     c_bool = None
 
 
+@record_function(FindMatchingRecord)
```

Comment:
my vote would be to define `FindMatchingRecord` in this file instead of `dbt_common.record` so that 'things that change together are live together'